contracts/PoolManager.sol

```diff
@@ -352,26 +351,26 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC1155, IERC1155Rec
         _accountDelta(currency, -(paid.toInt128()));
     }
 
-    function _burnAndAccount(Currency currency, uint256 amount) internal {
-        _burn(address(this), currency.toId(), amount);
-        _accountDelta(currency, -(amount.toInt128()));
-    }
-
-    function onERC1155Received(address, address, uint256 id, uint256 value, bytes calldata) external returns (bytes4) {
-        if (msg.sender != address(this)) revert NotPoolManagerToken();
-        _burnAndAccount(CurrencyLibrary.fromId(id), value);
-        return IERC1155Receiver.onERC1155Received.selector;
+    /// @inheritdoc IPoolManager
+    function burn(Currency currency, uint256 amount) external noDelegateCall onlyByLocker {
```

Comment:
Do we need `onlyByLocker` here?

Reply:
I actually think maybe not? burning only is only ever giving deltas, and the previous implementation didn't require sender to be locker

Reply:
Wouldn't it revert if called outside of a lock still since we do `getActiveLock`?